CONTRIBUTING.md

```diff
@@ -107,7 +121,47 @@ just ready
 This command will run the same commands of the CI: format, lint, tests and code generation.
 Eventually everything should be "green" 🟢 and commit all the code that was generated.
 
-## Language Server and VS Code Extension Development
+#### Generated files
+
+If you work on some parser and you create new nodes or modify existing ones, will need to run a command to update some files that are auto-generated.
+
+##### `cargo codegen grammar`
+
+This command will update the syntax of the parsers.
+
+The source is generated from the [`ungram` files](https://github.com/biomejs/biome/blob/main/xtask/codegen/js.ungram).
+
+##### `cargo codegen test`
+
+This command will create new tests for your parser. We currently have a neat infrastructure
+where tests for parser are generated com inline comments found inside
+the source code.
+
+
+##### `cargo codegen analyzer`
+
+This command will detect linter rules declared in the `analyzers` and `assists` directories in `biome_analyze`, regenerate the index modules `analyzers.rs` and `assists.rs` to import these files, and update the registry builder function in `registry.rs` to include all these rules.
+It will also regenerate the configuration of the rules.
```

Comment:
```suggestion
This command will detect linter rules declared in the `analyzers`, `assists` and `syntax` directories in the analyzer crates, e.g. `biome_js_analyze`, `biome_json_analyze`, etc., and regenerate the `registry.rs` file and its dependents to include all the rules.
```

The configuration file is not regenerated as part of this command. We have to run `cargo codegen-configuration` to do so.